src/Core/Lock/Lock.php

```diff
@@ -0,0 +1,69 @@
+<?php declare(strict_types=1);
+/**
+ * This File is part of JTL-Software
+ *
+ * User: avermeulen
+ * Date: 2021-01-26
+ */
+
+namespace JTL\SCX\Lib\Channel\Core\Lock;
+
+use JTL\SCX\Lib\Channel\Contract\Core\Lock\LockProvider;
+
+class Lock
+{
+    private LockProvider $lockProvider;
+    private string $key;
+    private int $ttl;
+    private bool $unlockOnDestruct;
+    private bool $isLocked;
+
+    public function __construct(LockProvider $lockProvider, string $key, int $ttl = 0, bool $unlockOnDestruct = true)
```

Comment:
Wollen wir die Gelegenheit nutzen und das Lock Object stateless gestalten?

d.h. $key wandert als Parameter in die einzelnen methoden